auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailActivity.java

```diff
@@ -118,17 +120,25 @@ public void onExistingIdpUser(User user) {
 
     @Override
     public void onNewUser(User user) {
-        // New user, direct them to create an account with email/password.
-        RegisterEmailFragment fragment = RegisterEmailFragment.getInstance(
-                mActivityHelper.getFlowParams(),
-                user);
-        FragmentTransaction ft = getSupportFragmentManager().beginTransaction()
-                .replace(R.id.fragment_register_email, fragment, RegisterEmailFragment.TAG);
+        // New user, direct them to create an account with email/password
+        // if account creation is enabled in SignInIntentBuilder
+
+        boolean createAccount = mActivityHelper.getFlowParams().createAccountIfEmailNotExists;
+        if (createAccount) {
 
-        View v = findViewById(R.id.email_layout);
-        if (v != null) ft.addSharedElement(v, "email_field");
+            RegisterEmailFragment fragment = RegisterEmailFragment.getInstance(
+                    mActivityHelper.getFlowParams(),
+                    user);
+            FragmentTransaction ft = getSupportFragmentManager().beginTransaction()
+                    .replace(R.id.fragment_register_email, fragment, RegisterEmailFragment.TAG);
 
-        ft.disallowAddToBackStack().commit();
+            View v = findViewById(R.id.email_layout);
+            if (v != null) ft.addSharedElement(v, "email_field");
+
+            ft.disallowAddToBackStack().commit();
+        } else {
+            finish(ResultCodes.NO_EMAIL_ACCOUNT, IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
```

Comment:
I think calling `finish()` here is a little severe, since the user could reach this state as a result of a typo.  Maybe we should just add an error to the email text field saying `No account exists for this email address` or something similar.